sphinx/application.py

```diff
@@ -298,10 +299,18 @@ def _init_builder(self, buildername):
         if buildername is None:
             print('No builder selected, using default: html', file=self._status)
             buildername = 'html'
-        if buildername not in self.builderclasses:
-            raise SphinxError('Builder name %s not registered' % buildername)
-
-        builderclass = self.builderclasses[buildername]
+        entry_points = iter_entry_points('sphinx.builders', buildername)
+        try:
+            entry_point = next(entry_points)
+        except StopIteration:
+            # fall back to old method
+            if buildername not in self.builderclasses:
+                raise SphinxError('Builder name %s not registered'
+                                  % buildername)
+            builderclass = self.builderclasses[buildername]
+        else:
+            builderclass = entry_point.load()
+            builderclass.setup(self)
```

Comment:
BTW builtin builders are already loaded. They might be loaded twice.
It would be better to load them only if needed. So I'd like to remove them from list of builtin_extensions.